crates/ty_python_semantic/src/types.rs

```diff
@@ -1761,6 +1772,11 @@ impl<'db> Type<'db> {
 
             (Type::Dynamic(_), _) | (_, Type::Dynamic(_)) => false,
 
+            (Type::TypedDict(_), _) | (_, Type::TypedDict(_)) => {
+                // TODO: Implement disjointness for TypedDict
+                false
```

Comment:
you could do something like this for now: if `Mapping[str, object]` is disjoint from the other type, any `TypedDict` type will also be disjoint from the other type, since all `TypedDict` types are subtypes of `Mapping[str, object]`

```suggestion
                KnownClass::Mapping.to_specialized_instance(db, [KnownClass::Str.to_instance(db), Type::object(db)]).is_disjoint_from(db, other)
```